### PR TITLE
[checkcommit] Fix logic for --skip-checkcommit and --force-checkcommit #86

### PR DESCRIPTION
**Pull request by PabloCastellano**
This check is run when any of the following conditions is true:

1. --force-checkcommit is specified
2. This function is invoked from Travis building a Pull Request

Context: https://github.com/openwisp/openwisp-utils/pull/90#issuecomment-609922762

Related to #86

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ install:
   - pip install -U -r requirements-test.txt
 
 before_script:
-  - openwisp-utils-qa-checks --migration-path "./tests/test_project/migrations"
+  - openwisp-utils-qa-checks --force-checkcommit --migration-path "./tests/test_project/migrations"
```
Comment (nemesisdesign) [inline .travis.yml:25]: let's not do this otherwise this build will fail at every merge

File: openwisp-utils-qa-checks [hunk 2]
```diff
@@ -83,15 +83,18 @@ runisort() {
     || { echoerr "ERROR: Isort check failed!"; FAILURE=1; }
 }
 
+# This check is run when any of the following conditions is true:
+# 1. --force-checkcommit is specified
+# 2. This function is invoked from Travis building a Pull Request
 runcheckcommit() {
-  [ -n "$TRAVIS_PULL_REQUEST" ] || TRAVIS_PULL_REQUEST="false"
-  if [ "$SKIP_CHECKCOMMIT" = "false" -o "$TRAVIS_PULL_REQUEST" != "false" ]; then
+	if [[ $FORCE_CHECKCOMMIT = true ]] ||
```
Comment (nemesisdesign) [inline openwisp-utils-qa-checks:90]: indentation is odd, can you put this at the same indentation level as it was before please?

File: openwisp-utils-qa-checks [hunk 3]
```diff
@@ -83,15 +83,18 @@ runisort() {
...
+		 [[ $TRAVIS = true && $TRAVIS_PULL_REQUEST = true ]]; then
```
Comment (nemesisdesign) [inline openwisp-utils-qa-checks:91]: what is `$TRAVIS = true`? Sounds like something that can break existing builds, please remove it.
We want to run this check only on pull requests for now.

File: openwisp-utils-qa-checks [hunk 4]
```diff
@@ -83,15 +83,18 @@ runisort() {
...
     echo "Running commit message check"
     checkcommit --message "$COMMIT_MESSAGE" \
     && echo "SUCCESS: Commit name check successful!" \
     || { printf "Checking commit message:\n\n"; printf "%s\n\n" "$COMMIT_MESSAGE"; echoerr "ERROR: Commit name check failed!"; FAILURE=1; }
   else
-    echo "SUCCESS: Commit name check successful!"
+    echo "SKIPPED: Commit name check"
```
Comment (nemesisdesign) [inline openwisp-utils-qa-checks:97]: spotted something odd in the old code, can you replace the instances of "Commit name" with "Commit message" please?

File: openwisp-utils-qa-checks [hunk 5]
```diff
@@ -83,15 +83,18 @@ runisort() {
     || { echoerr "ERROR: Isort check failed!"; FAILURE=1; }
 }
 
+# This check is run when any of the following conditions is true:
+# 1. --force-checkcommit is specified
+# 2. This function is invoked from Travis building a Pull Request
 runcheckcommit() {
-  [ -n "$TRAVIS_PULL_REQUEST" ] || TRAVIS_PULL_REQUEST="false"
```
Comment (nemesisdesign) [inline openwisp-utils-qa-checks:87]: I think this line was fine, I would not remove it

File: openwisp-utils-qa-checks (hunk 3 above)
Comment (PabloCastellano) [reply openwisp-utils-qa-checks]: `TRAVIS = true` when the script is being invoked by Travis (https://docs.travis-ci.com/user/environment-variables/#default-environment-variables) but you are right. Removed.

**Pull request merged**